src/server/game/Entities/Unit/Unit.h

```diff
@@ -2297,6 +2305,9 @@ class TC_GAME_API Unit : public WorldObject
         typedef std::list<GameObject*> GameObjectList;
         GameObjectList m_gameObj;
 
+        typedef std::list<AreaTrigger*> AreaTriggerList;
```

Comment:
@matukaa `std::remove` does not remove the element from container. It moves it to the end of it and gives you iterator pointing to it (more precisely, past the end of unerased elements = first erased one)